rlberry/utils/check_agent.py

```diff
@@ -222,6 +250,95 @@ def check_seeding_agent(agent, env=None, continuous_state=False, init_kwargs=Non
     assert result, "Agent not reproducible (same seed give different results)"
 
 
+def check_multi_fit(agent, env="continuous_state", init_kwargs=None):
+    """
+    Check that fitting two times with budget greater than n_step (buffer size) is working.
+
+    Parameters
+    ----------
+    agent: rlberry agent module
+        Agent class to test.
+    env: tuple (env_ctor, env_kwargs) or str in ["continuous_state", "discrete_state"], default="continuous_state"
+        if tuple, env is the constructor and keywords of the env on which to test.
+        if str in ["continuous_state", "discrete_state"], we use a default Benchmark environment.
+    init_kwargs : dict
+        Arguments required by the agent's constructor.
+    """
+    if init_kwargs is None:
+        init_kwargs = {}
+
+    init_kwargs["seeder"] = SEED
+    train_env_d = _make_env(env)
+    train_env = train_env_d[0](**train_env_d[1])
+
+    test_load_env_d = _make_env(env)
+    test_load_env = test_load_env_d[0](**test_load_env_d[1])
+
+    agent1 = agent(train_env, **init_kwargs)
+
+    if "n_steps" in agent1.get_params():
+        agent1.n_steps = 30
+
+    agent1.fit(100)
+    agent1.fit(100)
+
+    # test
+    state = test_load_env.reset()
+    for tt in range(50):
+        action = agent1.policy(state)
+        next_s, _, done, test = test_load_env.step(action)
+        if done:
+            break
+        state = next_s
+
+
+def check_vectorized_env_agent(
+    agent, env="vectorized_env_continuous", agent_init_kwargs=None
+):
+    """
+    Check that (multi-)fitting vectorized_env is working.
+
+    Parameters
+    ----------
+    agent: rlberry agent module
+        Agent class to test.
+    env: tuple (env_ctor, env_kwargs) or str in {"continuous_state", "discrete_state","vectorized"}, default="vectorized_env_continuous"
+        if tuple, env is the constructor and keywords of the env on which to test.
+        if str in {"continuous_state", "discrete_state","vectorized"}, we use a default Benchmark environment.
+    agent_init_kwargs : dict
+        Arguments required by the agent's constructor.
```

Comment:
I think you don't need env that can be continuous_state or discrete_state. It can only be vectorized_env_continuous and maybe a tuple.